### PR TITLE
Improved LitmusHttpClient structure with error handling

### PR DESCRIPTION
**Pull request by seedspirit**
I have made several improvements to our API client structure:

- Encapsulated HTTP response handling and parsing logic within LitmusHttpClient
- Introduced ApiException for more robust error handling
- Centralized host configuration by moving it to LitmusHttpClient constructor

These changes simplify API method implementations, improve type safety, and provide more consistent error handling

### CODE REVIEW

**Comment by seedspirit** [comment]
@namkyu1999 @suyeon-jung-dev @jemlog Can you please take a look?

**Comment by namkyu1999** [comment]
Can you check a DCO?

**Comment by seedspirit** [comment]
@namkyu1999 Thanks for notice! Applied signoff to the commits in the pull request and pushed again.

File: src/main/java/io/litmuschaos/LitmusClient.java [hunk 1]
```diff
@@ -28,23 +28,21 @@ public void close() throws Exception {
     }
 
     // TODO - @Suyeon Jung : host, port config to LitmusAuthConfig class
-    public LoginResponse authenticate(String host, String username, String password) throws IOException {
+    public LoginResponse authenticate(String username, String password) throws IOException {
         LoginRequest request = new LoginRequest(username, password);
-        LoginResponse response = httpClient.post(host + "/login", request, LoginResponse.class);
+        LoginResponse response = httpClient.post("/login", request, LoginResponse.class);
         this.token = response.getAccessToken();
         return response;
```
Comment (jemlog) [inline src/main/java/io/litmuschaos/LitmusClient.java:35]: In litmusClient constructor, we initialize token value by `credential.getAccessToken()` method. 
So, I think initializing part in `authenticate` method is not required. If authenticate method is used by client independently, I think removing initialize part in constructor is better solution.

Comment (seedspirit) [reply src/main/java/io/litmuschaos/LitmusClient.java:34]: Hi @jemlog
Thanks for the review. As authenticate method should served independtly, I've removed duplicated token initialization in `LitmusClient` constructor. 

```java
public LitmusClient(String host, String username, String password) throws IOException {
        this.httpClient = new LitmusHttpClient(host);
        this.authenticate(username, password);
    }
```

```java
public LoginResponse authenticate(String username, String password) throws IOException {
        LoginRequest request = new LoginRequest(username, password);
        LoginResponse response = httpClient.post("/login", request, LoginResponse.class);
        this.token = response.getAccessToken();
        return response;
    }
```

**Review by suyeon-jung-dev — COMMENTED** [review]
Sorry, this comment has been pending all this time. I've now submitted these reviews. Please check.

File: src/main/java/io/litmuschaos/http/LitmusHttpClient.java [hunk 2]
```diff
@@ -1,40 +1,101 @@
...
                 .post(body)
                 .header("Authorization", "Bearer " + token)
                 .build();
-        return transform(request, returnType);
+        try (Response response = okHttpClient.newCall(request).execute()) {
+            return handleResponse(response, responseType);
+        }
     }
 
     private <T> T transform(Request request, Class<T> returnType) throws IOException {
```
[the remark this answers is not in the document]
Comment (seedspirit) [reply src/main/java/io/litmuschaos/http/LitmusHttpClient.java]: @suyeon-jung-dev 

Thanks for the review. As you mentioned, I've moved `handleResponse` method in new `HttpResponseHandler` class. Also refactored `handleResponse` method to judge whether response is succeed or  failed, `transform` method to make Gson transformation

```java
public class HttpResponseHandler {

    private final Gson gson;

    public HttpResponseHandler(){
        this.gson = new GsonBuilder().create();
    }

    public <T> T handleResponse(Response response, Class<T> responseType) throws ApiException, IOException {
        if (!response.isSuccessful()) {
            throw new ApiException(response);
        }
        if (response.body() == null) {
            return transform("", responseType);
        }
        return transform(response.body().string(), responseType);
    }

    public <T> T handleResponse(Response response, Type responseType) throws ApiException, IOException {
        if (!response.isSuccessful()) {
            throw new ApiException(response);
        }
        if (response.body() == null) {
            return transform("", responseType);
        }
        return transform(response.body().string(), responseType);
    }

    private <T> T transform(String response, Class<T> responseType) {
        return gson.fromJson(response, responseType);
    }

    private <T> T transform(String response, Type responseType) {
        return gson.fromJson(response, responseType);
    }
}
```

**Comment by jemlog** [comment]
LGTM 🚀

**Comment by suyeon-jung-dev** [comment]
LGTM 🚀

**Pull request merged**